firestore/app/src/main/java/com/google/example/firestore/SolutionBundles.java

```diff
@@ -0,0 +1,85 @@
+package com.google.example.firestore;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.Continuation;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.LoadBundleTask;
+import com.google.firebase.firestore.LoadBundleTaskProgress;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.google.firebase.firestore.Source;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * https://firebase.google.com/docs/firestore/solutions/serve-bundles
+ */
+public class SolutionBundles {
+
+    private static final String TAG = "SolutionBundles";
+
+    private FirebaseFirestore db;
+
+    // [START fs_bundle_load]
+    public InputStream getBundleStream(String urlString) throws IOException {
+        URL url = new URL(urlString);
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        return connection.getInputStream();
+    }
+
+    public void fetchBunldeFrom() throws IOException {
+        final InputStream bundleStream = getBundleStream("https://example.com/createBundle");
+        LoadBundleTask loadTask = db.loadBundle(bundleStream);
+
+        // Chain the following tasks
+        // 1) Load the bundle
+        // 2) Get the named query from the local cache
+        // 3) Execute a get() on the named query
+        loadTask.continueWithTask(new Continuation<LoadBundleTaskProgress, Task<Query>>() {
+            @Override
+            public Task<Query> then(@NonNull Task<LoadBundleTaskProgress> task) throws Exception {
+                // Close the stream
+                bundleStream.close();
```

Comment:
Sam - do you think the SDK should close the stream? I am not sure what the expected behavior is.

Reply:
I was wondering the same thing ... I personally would like that but I really don't do enough Java I/O programming to have an informed opinion.